From 54bec089b735c1bb28d616364b661c6ad0e4b709 Mon Sep 17 00:00:00 2001
From: Jaren Johnston <jaren.johnston>
Date: Fri, 15 Jun 2012 15:45:28 +0200
Subject: [PATCH 214/292] Removed the guards against empty PT and empty AAD.
 It's an edge case that -- while it may not be too useful in practice -- is
 tested against during crypto validation.

Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>
---
 drivers/staging/crypto/cryptodev/authenc.c | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/drivers/staging/crypto/cryptodev/authenc.c b/drivers/staging/crypto/cryptodev/authenc.c
index 9ceee2d56c11..70f0f7be5341 100644
--- a/drivers/staging/crypto/cryptodev/authenc.c
+++ b/drivers/staging/crypto/cryptodev/authenc.c
@@ -730,17 +730,10 @@ int crypto_auth_run(struct fcrypt *fcr, struct kernel_crypt_auth_op *kcaop)
 	cryptodev_cipher_set_iv(&ses_ptr->cdata, kcaop->iv,
 				min(ses_ptr->cdata.ivsize, kcaop->ivlen));
 
-	if (likely(caop->len || caop->auth_len)) {
-		ret = __crypto_auth_run_zc(ses_ptr, kcaop);
-		if (unlikely(ret)) {
-			dprintk(1, KERN_ERR,
-				"error in __crypto_auth_run_zc()\n");
-			goto out_unlock;
-		}
-	} else {
+	ret = __crypto_auth_run_zc(ses_ptr, kcaop);
+	if (unlikely(ret)) {
 		dprintk(1, KERN_ERR,
-			"Do not have auth data nor other data.\n");
-		ret = -EINVAL;
+			"error in __crypto_auth_run_zc()\n");
 		goto out_unlock;
 	}
 
-- 
1.8.1.2

